### PR TITLE
Fix inconsistencies, allow ES6

### PR DESCRIPTION
**Pull request by AerialX**
jsfmt was kind of inconsistent with the other rules, and I'm not sure why we were using both it and jscs to begin with? I'd suggest moving on to eslint in the future.

Major changes here are the addition of `disallowMultipleVarDecl` and `requireSpacesInAnonymousFunctionExpression`.

### CODE REVIEW

**Comment by jeffheifetz** [comment]
At the time, jscs didn't do formatting. Eslint is definitely the way forward.

File: js/.jscsrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+    "requireSemicolons": true,
```
Comment (jeffheifetz) [inline js/.jscsrc]: I thought we were going with StandardJS?

Comment (AerialX) [inline js/.jscsrc]: This was more to be consistent with what the previous style defined (the jsfmtrc required semicolons). We can move away from them, I have no strong opinions either way.

**Comment by AerialX** [comment]
Besides semicolons I believe the main conflict here with standard js is their "space after function name" rule, which I personally think is awful but it may be the only difference.

File: js/.jscsrc (hunk 1 above)
Comment (HawkinH) [inline js/.jscsrc]: Additionally, I believe we had come to an agreement regarding the inconsistency of space usage in StandardJS in general

Comment (AerialX) [inline js/.jscsrc]: The issue with standard js is that it's underspecified, just a set of loose "don't do this" rules that don't create a full style guide. I'm not sure I understand the advantage of the project over a standardized (and more specific) eslint config. However, I can buy their semicolon argument; they are essentially redundant in javascript.

**Comment by jeffheifetz** [comment]
@AerialX Are we good with this?

**Comment by AerialX** [comment]
Yes but we need to merge https://github.com/BublTechnology/ScarletTests/pull/14 first

**Pull request merged**